### PR TITLE
🚀♻️ Extract `amp-story-interactive` host styles into extension

### PR DESCRIPTION
**Pull request by alanorozco**
The provided custom props are not required as part of the larger `amp-story-1.0.js` bundle. We move them into the `amp-story-interactive-1.0.js` bundle instead.

This shaves off `~0.55 kb` compressed from the larger bundle.

### CODE REVIEW

File: extensions/amp-story-interactive/0.1/amp-story-interactive-abstract.js
Comment (mszylkowski) [inline extensions/amp-story-interactive/0.1/amp-story-interactive-abstract.js:126]: I think it would work well if we kept the system `style[amp-extension="amp-story-interactive"]` to install the service, why are we doing this instead?

File: extensions/amp-story-interactive/0.1/amp-story-interactive-abstract.js
Comment (mszylkowski) [inline extensions/amp-story-interactive/0.1/amp-story-interactive-abstract.js:229]: I thought you were going to call this in `amp-story-interactive.js` before running the `registerElements` so it's guaranteed to be installed once, instead of trying to install it once per component.

Comment (alanorozco) [reply extensions/amp-story-interactive/0.1/amp-story-interactive-abstract.js]: I was following the behavior of `instalStylesOrDoc`, which verifies whether the style has already been added. 

Subjectively, it makes more sense to me to place the call in the abstract implementation instead of `AMP.extension()`.   It's uniform with shadow styles, and it mostly corresponds to the classnames set in the same file.

File: extensions/amp-story-interactive/0.1/amp-story-interactive-abstract.js
Comment (alanorozco) [reply extensions/amp-story-interactive/0.1/amp-story-interactive-abstract.js]: There are two options to use the existing `style[amp-extension="amp-story-interactive"]` system:

1. By bundling `installExtensionsForDoc()` as part of `amp-story-interactive`, which has a `~0.16 kb` impact on `amp-story-interactive-1.0.js`.

2. By placing the host CSS in every inherited component, which has a variable impact on runtime performance by overriding styles once for every tag name defined in the extension (10x).

I think the size tradeoff `amp-story-interactive-1.0.js` is more reasonable, so I chose to write a function to install once. However, the hand-rolled function is barely smaller than `installExtensionsForDoc()`, so I removed it and used the latter instead.

**Review by mszylkowski — APPROVED** [review]

**Review by gmajoulet — APPROVED** [review]
This is cool, let's ship it please!!

**Comment by mszylkowski** [comment]
@alanorozco whenever you get the chance, this PR is approved and (after conflict resolution) ready to merge

**Comment by gmajoulet** [comment]
Friendly ping on this, would love to ship it : ))

**Pull request closed without merging**
